githubmarkdownui/blocks/lists.py

```diff
@@ -10,6 +10,18 @@ class HtmlList(ABC):
     """
     items: List[Union[str, HtmlList]]
 
+    def build_list_contents(self) -> str:
+        """Constructs the contents of the list by building the <li> tags or any nested lists."""
+        list_body = ''
+
+        for item in self.items:
+            if isinstance(item, HtmlList):
+                list_body += str(item)
```

Comment:
so this isn't recursively calling `build_list_contents`?

Reply:
It's implicitly called through the `str(item)` call. We need the `<li>` or `<ol>` tags of the inner list here to format the list properly in HTML, and when we cast the list the `__str__` function will call `build_list_contents()`

Reply:
Do you think it'd be more clear to document in `HtmlList` that, `__str__` needs to be overridden to output HTML-flavored markdown syntax in inherited classes?

Reply:
Good idea. I've updated the docstring of `HtmlList` to state that, and also marked `__str__` as an abstract method